contexts/ErrorContext.tsx

```diff
@@ -1,34 +1,44 @@
 import { createContext, ReactNode, useState } from "react";
 
 export interface ErrorContextState {
-  errorMessages: ReactNode[];
-  addErrorMessage: (message: ReactNode) => void;
-  removeErrorMessage: (message: ReactNode) => void;
-  clearErrorMessages: () => void;
+  errorMessages: Record<string,ReactNode[]>;
```

Comment:
nit: formatting in `string,ReactNode` will make the build fail. I've fixed this issue with prettier not applying in a pending PR